web/modules/custom/joinup_core/joinup_core.module

```diff
@@ -201,3 +211,56 @@ function joinup_core_module_implements_alter(array &$implementations, $hook) {
       break;
   }
 }
+
+/**
+ * Implements hook_block_access().
+ *
+ * Determines on which pages to show the Webtools eTrans block. Not all pages
+ * are suitable for machine translation.
+ *
+ * This is handled using a custom access check rather than configuring block
+ * visibility conditions in the block UI since we need to check conditions
+ * across several domains. This is not possible in the UI because the conditions
+ * are compared using "AND" logic.
+ *
+ * @see https://www.drupal.org/project/drupal/issues/923934
+ */
+function joinup_core_block_access(BlockInterface $block, string $operation, AccountInterface $account): AccessResultInterface {
```

Comment:
It would have been nice and elegant if the whole logic here was a _visibility condition_.

Reply:
I was in doubt about this. A visibility condition is available to all blocks and cannot be hidden from the UI. Since this is hardcoded logic I wanted to limit it only to this block.

Reply:
I discussed this today with @claudiu-cristea and we decided to keep the hook.